xarray/core/treenode.py

```diff
@@ -560,7 +560,10 @@ def __delitem__(self: Tree, key: str):
             del self._children[key]
             child.orphan()
         else:
-            raise KeyError("Cannot delete")
+            try:
+                raise ValueError("cannot delete key not found in child nodes")
+            except ValueError as e:
+                raise KeyError(key) from e
```

Comment:
This is a partial work-around for:
https://github.com/pydata/xarray/issues/9221

Reply:
So this is just a weird way to get a KeyError raised from a ValueError?

Reply:
Yes, you get an error message like this:
```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
File ~/dev/xarray/xarray/core/treenode.py:564, in TreeNode.__delitem__(self, key)
    563 try:
--> 564     raise ValueError("cannot delete key not found in child nodes")
    565 except ValueError as e:

ValueError: cannot delete key not found in child nodes

The above exception was the direct cause of the following exception:

KeyError                                  Traceback (most recent call last)
Cell In[2], line 1
----> 1 del tree['x']

File ~/dev/xarray/xarray/core/treenode.py:566, in TreeNode.__delitem__(self, key)
    564     raise ValueError("cannot delete key not found in child nodes")
    565 except ValueError as e:
--> 566     raise KeyError(key) from e

KeyError: 'x'
```

Reply:
Why not simply `raise KeyError(key) from ValueError("cannot delete key not found in child nodes")`?

Reply:
> Why not simply `raise KeyError(key) from ValueError("cannot delete key not found in child nodes")`?

That would definitely be a better way to write this! Not sure why I didn't think of it